main.go

```diff
@@ -341,30 +353,57 @@ func main() {
 		log.Donef("Using system installed CocoaPods version")
 	}
 
-	// Check ruby version
-	// Run this logic only in CI environment when the ruby was installed via rbenv for the virtual machine
-	if os.Getenv("CI") == "true" && rubycommand.RubyInstallType() == rubycommand.RbenvRuby {
-		fmt.Println()
-		log.Infof("Check selected Ruby is installed")
-
+	if rubycommand.RubyInstallType() == rubycommand.RbenvRuby {
+		rubySelectStart := time.Now()
 		rubyInstalled, rversion, err := rubycommand.IsSpecifiedRbenvRubyInstalled(configs.SourceRootPath)
 		if err != nil {
-			log.Errorf("Failed to check if selected ruby is installed, error: %s", err)
+			log.Errorf("Failed to check if selected ruby is installed: %s", err)
 		}
 
-		if !rubyInstalled {
-			log.Errorf("Ruby %s is not installed", rversion)
+		// Check ruby version
+		// Run this logic only in CI environment when the ruby was installed via rbenv for the virtual machine
+		if os.Getenv("CI") == "true" {
 			fmt.Println()
+			log.Infof("Check selected Ruby is installed")
+
+			if !rubyInstalled {
+				log.Errorf("Ruby %s is not installed", rversion)
+				fmt.Println()
 
-			cmd := command.New("rbenv", "install", rversion).SetStdout(os.Stdout).SetStderr(os.Stderr)
-			log.Donef("$ %s", cmd.PrintableCommandArgs())
-			if err := cmd.Run(); err != nil {
-				log.Errorf("Failed to install Ruby version %s, error: %s", rversion, err)
+				cmd := command.New("rbenv", "install", rversion).SetStdout(os.Stdout).SetStderr(os.Stderr)
+				log.Donef("$ %s", cmd.PrintableCommandArgs())
+				if err := cmd.Run(); err != nil {
+					log.Errorf("Failed to install Ruby version %s, error: %s", rversion, err)
+				}
+			} else {
+				log.Donef("Ruby %s is installed", rversion)
 			}
-		} else {
-			log.Donef("Ruby %s is installed", rversion)
 		}
 
+		rubySelectDuration := time.Since(rubySelectStart)
+		isRequiredRubyInstalled, _, err := rubycommand.IsSpecifiedRbenvRubyInstalled(podfileDir)
+		if err != nil {
+			log.Errorf("Failed to check if selected ruby is installed: %s", err)
+		}
+
+		effectiveRubyVersion, err := command.New("rbenv", "global").RunAndReturnTrimmedOutput()
+		if err != nil {
+			log.Errorf("Failed to check global rbenv version: %w", err)
+		}
+		if isRequiredRubyInstalled {
+			effectiveRubyVersion = rversion
+		}
+
+		log.Infof("Sending analytics...")
+		tracker.Enqueue("step_ruby_version_selected", analytics.Properties{
+			"step_execution_id":         envRepository.Get("BITRISE_STEP_EXECUTION_ID"),
+			"build_slug":                envRepository.Get("BITRISE_BUILD_SLUG"),
+			"step_id":                   "cocoapods-install",
+			"requested_ruby_version":    rversion,
+			"effective_ruby_version":    effectiveRubyVersion,
+			"version_change_duration_s": int64(rubySelectDuration.Seconds()),
+		})
+		defer tracker.Wait()
```

Comment:
I'd rather call this at the end of the main() function (or any other high-level function), so that:
- it's only called once even if there are multiple enqueue() calls
- it doesn't block the step execution in the middle unnecessarily, this enqueue call has time to finish the request until the end of the main block execution.

Reply:
It would ran on the end of the main function technically (https://stackoverflow.com/questions/49456943/why-is-golang-defer-scoped-to-function-not-lexical-enclosure), but I agree that meybe it is not in the best place.